File: app/src/main/java/com/lazylee/lzywanandroid/activity/register/RegisterContract.java
package com.lazylee.lzywanandroid.activity.register;

import com.lazylee.lzywanandroid.mvp.BasePresenter;
import com.lazylee.lzywanandroid.mvp.BaseView;

/**
 * register contract
 * Created by lazylee on 2018/3/22.
 */

public interface RegisterContract {

    interface View extends BaseView<Presenter> {

        void enableRegisterButton(boolean enable);

        void showUserNameError(String error);

        void showPasswordError(String error);

        void showRepasswordError(String error);

        void showProgressBar(boolean show);
    }

    interface Presenter extends BasePresenter {

        void register(String username, String password, String repassword);
    }
}

File: app/src/main/java/com/lazylee/lzywanandroid/ui/activity/register/RegisterContract.kt
package com.lazylee.lzywanandroid.ui.activity.register

import com.lazylee.lzywanandroid.mvp.BasePresenter
import com.lazylee.lzywanandroid.mvp.BaseView

/**
 * register contract
 * Created by lazylee on 2018/3/22.
 */

interface RegisterContract {

    interface View : BaseView<Presenter> {

        fun enableRegisterButton(enable: Boolean)

        fun showUserNameError(error: String)

        fun showPasswordError(error: String)

        fun showRepasswordError(error: String)

        fun showProgressBar(show: Boolean)
    }

    interface Presenter : BasePresenter {

        fun register(username: String, password: String, repassword: String)
    }
}
